add-on/src/lib/blockchain-domains.js

```diff
@@ -0,0 +1,68 @@
+const { Resolution } = require('@unstoppabledomains/resolution')
+const browser = require('webextension-polyfill')
+// Main tool to resolve blockchain domains
+const resolution = new Resolution({
+  blockchain: {
+    ens: { url: 'https://mainnet.infura.io/v3/7253dea473954208bede79ae26ae1225' },
+    cns: { url: 'https://mainnet.infura.io/v3/7253dea473954208bede79ae26ae1225' }
+  }
+})
```

Comment:
It is possible to use any public provider similar to infura. Which gateways do you want me to include? infura link here plays just an example of what is possible.

Reply:
If provider can be customized, then there should be UI where user can pick a different one. Ideally, on the same screen where user gives consent for sending requests to third party API + at any time on _Preferences_ screen.

I am not familiar with this API: who else is implementing it?

Reply:
Infura is playing a role of a public gateway to blockchain. It is an etherium node that opens it's gate to anyone who wishes to communicate with smart contracts without launching the etherium node for themselves. If an end user has an etherium node set up he could delicate the resolution library towards it by using something like localhost:port url there. In case if ipfs-companion has an etherium node launched we can set it up so the request goes through them. I am not very familiar with other solutions on public etherium gateway other than infura. 

The drawback is that Infura does require each request send with the project id ( last number after v3 ) you can obtain on registering for infura project on their website https://infura.io. After some rate the project is forced to upgrade to paid membership that's how they make their revenue. 

I will create a possibility for an end user to provide their provider url or choose to use infura. The infura project id I chose for this or is newly registered and will require some updates later from ipfs-companion side to have full control of the statistics and ability to upgrade later if needed.

Reply:
Thank you for this context. If I understand this correctly, using HTTP API from Infura for resolving `.crypto` and any other TLD comes with serious downsides:

- there is no other provider that implements this specific API (that we are aware of) 
  (Infura is not "a gateway" but "the gateway")
- Infura requires API key that will:
  - identify users of IPFS Companion  
  - force IPFS Companion to upgrade to paid membership if our users go over the free quota 
- there is no easy way to run localhost resolver that exposes drop-in API that could be used instead of Infura (or are there docs / pocs that do that?)

(Let me know if I got any of this wrong)